File: Java/PrimerParcial/Taller/subway/FixedSubwayCard.java
package ar.itba.edu.POO.PrimerParcial.Taller.subway;

public class FixedSubwayCard extends SubwayCard {
    private int rides;

    public FixedSubwayCard(SubwayCentral subwayCentral, int rides) {
        super(subwayCentral);
        this.rides = rides;
    }

    @Override
    protected boolean canRide() {
        return rides > 0;
    }

    @Override
    public void ride() {
        if(!canRide()) {
            System.err.println("No more trips.");
        }
        rides--;
    }
}

File: Ruby/TP8/Ej6/fixed_subway_card.rb
class FixedSubwayCard<SubwayCard
  def initialize(sub_central,trips_amount)
    super(sub_central)
    @trips_amount=trips_amount
  end
  def can_ride
    if @trips_amount<=0
      raise RuntimeError,"No rides Available"
    end
    true
  end
  def extract_trip
    @trips_amount-=1
  end

  def to_s
    "Fixed Subway Card ["+@subway_central.to_s+"][Rides: "+@trips_amount.to_s+"]"
  end
end
